File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        def getPalin(s,i1, i2):
            count = 0
            while i1 >= 0 and i2 < len(s) and s[i1] == s[i2]:
                count += 1
                i1 -= 1
                i2 += 1
            return count
        count = 0
        for i in range(len(s)):
            count += getPalin(s,i,i)
            count += getPalin(s,i,i + 1)
        return count


        

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> map = new HashMap<>();
        
        for (String str : strs){
            int[] count = new int[26];
            for (char c : str.toCharArray()) {
                count[c - 'a'] += 1;
            }
            String c = Arrays.toString(count);
            if (map.get(c) == null) map.put(c, new ArrayList<String>());
            map.get(c).add(str);
        }
        List<List<String>> ans = new ArrayList<>(map.values());
        return ans;
    }
}
